lib/ansible/galaxy/token.py

```diff
@@ -54,11 +59,19 @@ def __init__(self, access_token=None, auth_url=None, validate_certs=True, client
         self.access_token = access_token
         self.auth_url = auth_url
         self._token = None
+        self._issued_at = None
+        self._expires_in = None
         self.validate_certs = validate_certs
         self.client_id = client_id
         if self.client_id is None:
             self.client_id = 'cloud-services'
 
+    @property
+    def expired(self):
+        issue_time = datetime.strptime(self._issued_at, ISO_8601_FORMAT_STR)
```

Comment:
I think Python 3 has functions that already use ISO 8601 format so no need for a constant: https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat. Would this work?

Reply:
Python-specific ISO 8601 doesn't seem to handle the timezone.
```python
>>> datetime.fromisoformat("2022-11-27T13:50:51.925209Z")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid isoformat string: '2022-11-27T13:50:51.925209Z'
>>> datetime.strptime("2022-11-27T13:50:51.925209Z", "%Y-%m-%dT%H:%M:%S.%f%z").tzinfo
datetime.timezone.utc
>>> datetime.fromisoformat("2022-11-27T13:50:51.925209").tzinfo
>>>
```

Reply:
Hm... I seem to recall addressing this somehow. Let me see.

Reply:
Aha! `Z` is sometimes ambiguous: https://en.wikipedia.org/wiki/ISO_8601#Coordinated_Universal_Time_(UTC). Maybe that's why.

FWIW here's the solution I used in octomachinery: https://github.com/sanitizers/octomachinery/blob/94c3eaa7/octomachinery/github/models/utils.py#L27-L31.

Reply:
Specifically this works:
```pycon
In [9]: datetime.fromisoformat("2022-11-27T13:50:51.925209+00:00").tzinfo
Out[9]: datetime.timezone.utc
```

Reply:
Thanks for the pointer about Z, TIL. The format from AH is an integer timestamp which made this simpler.